src/support/ScrollBar.ts

```diff
@@ -43,139 +45,146 @@ export interface ScrollBarProperties extends ThemeableProperties {
+	 * @param event The pointer event
+	 */
+	private _onpointerleave(event: PointerEvent) {
+		event.preventDefault();
 		if (this._visible) {
 			this._visible = false;
 			this.invalidate();
 		}
 	}
 
-	private _onSliderDragStart = (evt: MouseEvent & TouchEvent) => {
-		evt.preventDefault();
-		const { horizontal } = this.properties;
-		this._dragging = true;
-		this._dragPosition = getAbsolutePosition(evt, horizontal);
-	}
-
-	private _onSliderDrag = (evt: MouseEvent & TouchEvent) => {
+	render() {
+		const domSize = this._getDomSize();
 		const {
-			_dragging,
-			_dragPosition,
 			properties: {
-				horizontal,
+				horizontal = false,
+				key = DEFAULT_KEY,
+				position,
+				size = domSize,
+				sliderMin = 10,
+				sliderSize = 0,
+				visible: propsVisible,
 				onScroll
 			}
 		} = this;
-		if (_dragging) {
-			evt.preventDefault();
-			const delta = this._toRelative(getAbsolutePosition(evt, horizontal) - _dragPosition);
-			onScroll && onScroll(delta);
-			this._dragPosition = getAbsolutePosition(evt, horizontal);
-		}
-	}
-
-	private _onSliderDragStop = (evt: MouseEvent & TouchEvent) => {
-		evt.preventDefault();
-		this._dragging = false;
-	}
-
-	public onElementCreated(element: HTMLElement, key: string) {
-		this._onDomUpdate(element, key);
-	}
 
-	public onElementUpdated(element: HTMLElement, key: string) {
-		this._onDomUpdate(element, key);
-	}
+		let dragging = false;
+		const dragResult = this.meta(Drag).get('slider');
+		const delta = horizontal ? dragResult.delta.x : dragResult.delta.y;
+		dragging = dragResult.isDragging;
+		delta && onScroll(toRelative(delta, size, domSize));
 
-	render() {
-		const {
-			horizontal = false,
-			key = DEFAULT_KEY,
-			position,
-			size = 0,
-			sliderMin = 10,
-			sliderSize = 0,
-			visible: propsVisible
-		} = this.properties;
-		const renderPosition = String(this._fromRelative(position)) + 'px';
-		const relativeSliderSize = this._fromRelative(sliderSize);
-		const renderSliderSize = String(relativeSliderSize > sliderMin ? relativeSliderSize : sliderMin) + 'px';
+		const renderPosition = `${fromRelative(position, size, domSize)}px`;
+		const absoluteSliderSize = fromRelative(sliderSize, size, domSize);
+		const renderSliderSize = `${absoluteSliderSize > sliderMin ? absoluteSliderSize : sliderMin}px`;
 		const visible = sliderSize >= size ? false : propsVisible !== undefined ? propsVisible : this._visible;
-		const styles = horizontal ? {
-			left: renderPosition,
-			width: renderSliderSize
-		} : {
-			top: renderPosition,
-			height: renderSliderSize
+		const styles = {
+			[ horizontal ? 'left' : 'top' ]: renderPosition,
+			[ horizontal ? 'width' : 'height' ]: renderSliderSize
 		};
 
 		return v('div', {
-			classes: this.classes(css.root, css.vertical, visible ? css.visible : css.invisible),
+			classes: this.classes(
+				css.root,
+				horizontal ? css.horizontal : css.vertical,
+				visible || dragging ? css.visible : css.invisible
```

Comment:
Not sure what the style guide says, but I think `( visible || dragging )` would read easier than `visible || dragging`